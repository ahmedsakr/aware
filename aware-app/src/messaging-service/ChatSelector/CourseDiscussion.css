#vertical-menu {
    background-color: #18202c;
    display: block;
    text-align: left;
    height: 100%;
}

#vertical-menu > h3 {
  color: white;
  padding-left: 10px;
}

.chat-navigation-item {
    display: block;
    margin: 0px;
}

.navbar-item-name, .navbar-item-avatar {
  display: inline-block;
}

.navbar-item-name {
  color:white;
}

.chat-navigation-item:hover {
    background-color: #232f3e;
    text-decoration: none;
}

.chat-navigation-item.active {
    background-color: #4CAF50;
    color: white;
}

#tag:hover {
  background-color: #232f3e;
}

hr {
  background-color: white;
  height: 3;
}
