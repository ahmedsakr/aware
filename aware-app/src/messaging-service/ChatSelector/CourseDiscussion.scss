/*
 * Course Discussion styling
 *
 * Defines layout fundamentals for Course Discussion.
 */

$chat_navigation_background_color: #232f3e;
$chat_navigation_active_color: #4CAF50;
$navbar_item_color: #babdc0;

.chat-navigation-item {
    display: block;
    margin: 0px;
}

.chat-navigation-item-selected {
    display: block;
    margin: 0px;
    background-color: $chat_navigation_background_color;
}

.navbar-item-name,
.navbar-item-avatar {
    display: inline-block;
}

.navbar-item-name {
    color: $navbar_item_color;

    >p {
        margin: 0px;
    }
}

.chat-navigation-item {
    &:hover {
        background-color: $chat_navigation_background_color;
        text-decoration: none;
    }

    &.active {
        background-color: $chat_navigation_active_color;
        color: white;
    }
}

#tag:hover {
    background-color: $chat_navigation_background_color;
}

.chat-selector-line-break {
    background-color: white;
    height: 3;
    width: 100%;
}