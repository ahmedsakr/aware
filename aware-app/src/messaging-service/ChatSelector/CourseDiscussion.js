import React, {Component} from 'react';
import './CourseDiscussion.css'
class CourseDiscussion extends Component {
  render() {
    const {src, name} = this.props;

    return (
      <a href="/" class="chat-navigation-item">
        <div class="navbar-item-avatar col-2">
          <img src={process.env.PUBLIC_URL + src} alt={name}/>
        </div>
        <div class="navbar-item-name col-9">
          <p>{name.toUpperCase()}</p>
        </div>
      </a>
    );
  }
}

export default CourseDiscussion;
