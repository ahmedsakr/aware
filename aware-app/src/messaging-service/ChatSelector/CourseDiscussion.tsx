import React from 'react';
import './CourseDiscussion.scss'

type CourseDiscussionProps = {
    updateRoom: (course: CourseDiscussion) => void,
    room: string,
    src: string,
    name: string,
};

type CourseDiscussionState = {
    selected: boolean
};

export default class CourseDiscussion extends React.Component<CourseDiscussionProps, CourseDiscussionState> {
    constructor(props: CourseDiscussionProps) {
        super(props)

        this.state = {
            selected: false
        }
    }

    shouldComponentUpdate(nextProps: CourseDiscussionProps, nextState: CourseDiscussionState): boolean {
        return nextState.selected !== this.state.selected;
    }

    render(): JSX.Element {
        const { src, name, room, updateRoom } = this.props;

        return (
            <div onClick={() => { updateRoom(this); }} className={this.state.selected ? "chat-navigation-item-selected" : "chat-navigation-item"}>
                <div className="navbar-item-avatar col-2">
                    <img src={process.env.PUBLIC_URL + src} alt={name} />
                </div>
                <div className="navbar-item-name col-9">
                    <p>{name}</p>
                </div>
            </div>
        );
    }
}
