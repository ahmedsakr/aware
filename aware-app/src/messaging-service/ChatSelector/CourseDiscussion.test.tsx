import React from 'react';
import { shallow } from 'enzyme';

import CourseDiscussion from './CourseDiscussion';

describe('CourseDiscussion', () => {
    it('should render correctly', () => {

        // Stub out the function for now, but we probably need to
        // implement better tests in the future.
        let updateRoom = (course: CourseDiscussion) => { };

        const component = shallow(
            <CourseDiscussion
                updateRoom={updateRoom}
                room="sysc2100"
                src="my-image.png"
                name="SYSC 2100" />
        );

        expect(component).toMatchSnapshot();
    });
});
