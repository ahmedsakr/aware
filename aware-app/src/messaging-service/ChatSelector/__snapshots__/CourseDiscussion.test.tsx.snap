version https://git-lfs.github.com/spec/v1
oid sha256:f579599b777be228641c04631df561c16880d100b2b54a0928a1d71b683d2506
size 393
